app/src/main/java/org/oppia/app/parser/StringToFractionParser.kt

```diff
@@ -1,13 +1,75 @@
 
+import org.oppia.app.customview.interaction.FractionInputInteractionView
 import org.oppia.app.model.Fraction
+import org.oppia.app.topic.FractionParsingErrors
 import org.oppia.domain.util.normalizeWhitespace
+import java.lang.Integer.parseInt
+import java.util.regex.Pattern
 
 /** This class contains method that helps to parse string to fraction. */
 class StringToFractionParser {
   private val wholeNumberOnlyRegex = """^-? ?(\d+)$""".toRegex()
   private val fractionOnlyRegex = """^-? ?(\d+) ?/ ?(\d+)$""".toRegex()
+  private val partialFractionOnlyRegex = """^-? ?(\d+) ?/ ?$""".toRegex()
   private val mixedNumberRegex = """^-? ?(\d+) (\d+) ?/ ?(\d+)$""".toRegex()
+  private val partialMixedNumberRegexFirst = """^-? ?(\d+) (\d+) ?$""".toRegex()
+  private val partialMixedNumberRegexSecond = """^-? ?(\d+) (\d+) ?/ ?$""".toRegex()
+  private val invalidCharsRegex = """^[\d\s/-]+$""".toRegex()
+  /**
+   * @param inputText is the user input in the [FractionInputInteractionView]
+   * This method helps to validate the inputText and return [FractionParsingErrors]
+   * This called on text change.
+   */
+  fun checkForErrors(inputText: String): FractionParsingErrors {
+    var rawInput: String = inputText.normalizeWhitespace()
+    if (!rawInput.matches(invalidCharsRegex))
+      return FractionParsingErrors.INVALID_CHARS
+    if (!rawInput.equals("-") && wholeNumberOnlyRegex.matchEntire(rawInput) == null &&
+      fractionOnlyRegex.matchEntire(rawInput) == null && partialFractionOnlyRegex.matchEntire(rawInput) == null &&
+      mixedNumberRegex.matchEntire(rawInput) == null && partialMixedNumberRegexFirst.matchEntire(rawInput) == null &&
+      partialMixedNumberRegexSecond.matchEntire(rawInput) == null
+    ) {
+      return FractionParsingErrors.INVALID_FORMAT
+    }
+    return FractionParsingErrors.VALID
+  }
+
+//TODO(#377): use this method check value valid and activate submit button click
+
+  /**
+   * @param inputText is the user input in the [FractionInputInteractionView]
+   * This method helps to validate the inputText and return [FractionParsingErrors]
+   * This called on submit button click.
+   */
+  fun checkForErrorsOnSubmit(inputText: String): FractionParsingErrors {
+    var denominator = 1
+    var rawInput: String = inputText.normalizeWhitespace()
+    if (!rawInput.matches(invalidCharsRegex))
+      return FractionParsingErrors.INVALID_CHARS
+    if (mixedNumberRegex.matchEntire(rawInput) == null && fractionOnlyRegex.matchEntire(rawInput) == null &&
+      wholeNumberOnlyRegex.matchEntire(rawInput) == null
+    ) {
+      return FractionParsingErrors.INVALID_FORMAT
+    }
+    rawInput = rawInput.trim();
+    if (rawInput.startsWith("-")) {
+      // Remove the negative char from the string.
+      rawInput = rawInput.substring(1).trim()
+    }
+    // Filter result from split to remove empty strings.
+    var numbers = Pattern.compile("[\\s|/]+").split(rawInput)
+    if (numbers.size == 1) {
```

Comment:
Android suggestion - Replace this with `when`

Reply:
Done